### PR TITLE
lxd-metadata: Annotate codebase for instance properties

### PR DESCRIPTION
**Pull request by gabrielmougard**
JIRA ticket: https://warthogs.atlassian.net/jira/software/c/projects/LXD/boards/54?assignee=63d9eb6128cddcc70770aff1&selectedIssue=LXD-439

### CODE REVIEW

**Comment by gabrielmougard** [comment]
@ru-fu this is the last doc PR to review for the config options :)

**Comment by tomponline** [comment]
please rebase

File: lxd/instance/drivers/driver_lxc.go [hunk 1]
```diff
@@ -6588,8 +6588,20 @@ func (d *lxc) templateApplyNow(trigger instance.TemplateTrigger) error {
...
+	// lxdmeta:generate(entities=instance-property; group=instance-conf; key=name)
+	// See {ref}`instance-name-requirements`
+	// ---
+	//  type: string
+	//  readonly: yes
+	//  shortdesc: Instance name
 	containerMeta["name"] = d.name
 	containerMeta["type"] = "container"
+	// lxdmeta:generate(entities=instance-property; group=instance-conf; key=architecture)
+	// See {ref}`instance-name-requirements`
```
Comment (ru-fu) [inline lxd/instance/drivers/driver_lxc.go:6600]:
```suggestion
```

File: lxd/instance/drivers/driver_lxc.go [hunk 2]
```diff
@@ -6588,8 +6588,20 @@ func (d *lxc) templateApplyNow(trigger instance.TemplateTrigger) error {
 
 	// Generate the container metadata
 	containerMeta := make(map[string]string)
+	// lxdmeta:generate(entities=instance-property; group=instance-conf; key=name)
+	// See {ref}`instance-name-requirements`
```
Comment (ru-fu) [inline lxd/instance/drivers/driver_lxc.go:6592]:
```suggestion
	// See {ref}`instance-name-requirements`.
```

**Review by ru-fu — CHANGES_REQUESTED** [review]
Small changes, so it's probably easier if you do them directly. ;)

File: lxd/instance/drivers/driver_lxc.go (hunk 2 above)
Comment (gabrielmougard) [reply lxd/instance/drivers/driver_lxc.go]: fixed

File: lxd/instance/drivers/driver_lxc.go (hunk 1 above)
Comment (gabrielmougard) [reply lxd/instance/drivers/driver_lxc.go]: fixed

Comment (ru-fu) [reply lxd/instance/drivers/driver_lxc.go]: This is still included?

File: lxd/instance/drivers/driver_lxc.go [hunk 3]
```diff
@@ -6588,8 +6588,20 @@ func (d *lxc) templateApplyNow(trigger instance.TemplateTrigger) error {
...
+	// lxdmeta:generate(entities=instance-property; group=instance-conf; key=name)
+	// See {ref}`instance-name-requirements`.
+	// ---
+	//  type: string
+	//  readonly: yes
+	//  shortdesc: Instance name
 	containerMeta["name"] = d.name
 	containerMeta["type"] = "container"
+	// lxdmeta:generate(entities=instance-property; group=instance-conf; key=architecture)
+	// See {ref}`instance-name-requirements`.
```
Comment (ru-fu) [inline lxd/instance/drivers/driver_lxc.go:6600]: This is wrong.

**Review by ru-fu — CHANGES_REQUESTED** [review]

File: lxd/instance/drivers/driver_lxc.go (hunk 1 above)
Comment (gabrielmougard) [reply lxd/instance/drivers/driver_lxc.go]: oh , sorry for the mistake. I'm removing it.

Comment (gabrielmougard) [reply lxd/instance/drivers/driver_lxc.go]: ok it should be removed now

**Review by ru-fu — APPROVED** [review]

**Review by tomponline — APPROVED** [review]

**Pull request merged**